libevmasm/RuleList.h

```diff
@@ -390,6 +390,27 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	{
+		SimplificationRule<Pattern> pat = {
+			// A & ((2^256-1) >> B) == ((2^256-1) >> B)
```

Comment:
This does not really describe the rule. It would probably be better to use `A & (X >> B) -> X >> B` and then state the condition.

Reply:
I was going with what @ekpyron suggested in https://github.com/ethereum/solidity/pull/6160#discussion_r265683822
But I can change it again, sure.